frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/backgroundables/RebaseOnParentBackgroundable.java

```diff
@@ -86,73 +92,77 @@ private GitRebaseParams getIdeaRebaseParamsOf(GitRepository repository, IGitReba
-
-      return;
-    }
-
-    final AtomicReference<Try<@Nullable IExecutionResult>> wrapper = new AtomicReference<>(
-        Try.success(null));
-    new GitFreezingProcess(project, getTitle(), () -> {
-      LOG.info("Executing machete-pre-rebase hooks");
-      Try<@Nullable IExecutionResult> hookResult = Try
-          .of(() -> gitMacheteRepositorySnapshot.executeMachetePreRebaseHookIfPresent(gitRebaseParameters));
-      wrapper.set(hookResult);
-    }).execute();
-    Try<@Nullable IExecutionResult> hookResult = wrapper.get();
-    if (hookResult == null) {
-      // Not really possible, it's here just to calm down Checker Framework.
-      return;
-    }
-
-    if (hookResult.isFailure()) {
-      val message = "machete-pre-rebase hooks refused to rebase ${NL}error: ${hookResult.getCause().getMessage()}";
-      LOG.error(message);
-      VcsNotifier.getInstance(project).notifyError(/* displayId */ null,
-          getString("action.GitMachete.BaseSyncToParentByRebaseAction.notification.title.rebase-abort"),
-          message);
-      return;
-    }
-
-    val executionResult = hookResult.get();
-    if (executionResult != null && executionResult.getExitCode() != 0) {
-      val message = "machete-pre-rebase hooks refused to rebase (exit code ${executionResult.getExitCode()})";
-      LOG.error(message);
-      val stdoutOption = executionResult.getStdout();
-      val stderrOption = executionResult.getStderr();
-      VcsNotifier.getInstance(project).notifyError(/* displayId */ null,
-          getString("action.GitMachete.BaseSyncToParentByRebaseAction.notification.title.rebase-abort"), message
-              + (!stdoutOption.isBlank() ? NL + "stdout:" + NL + stdoutOption : "")
-              + (!stderrOption.isBlank() ? NL + "stderr:" + NL + stderrOption : ""));
-      return;
-    }
-
-    val params = getIdeaRebaseParamsOf(gitRepository, gitRebaseParameters);
-    LOG.info("Rebasing '${gitRebaseParameters.getCurrentBranch().getName()}' branch " +
-        "until ${gitRebaseParameters.getForkPointCommit().getHash()} commit " +
-        "onto ${gitRebaseParameters.getNewBaseBranch().getName()}");
-
-    /*
-     * Git4Idea ({@link git4idea.rebase.GitRebaseUtils#rebase}) does not allow rebasing in detached head state. However, it is
-     * possible with Git (performing checkout implicitly) and should be allowed in the case of "Checkout and Rebase Onto Parent"
-     * Action. To pass the git4idea check in such a case, we checkout the branch explicitly and then perform the actual rebase.
-     */
-    if (shouldExplicitlyCheckout) {
-      val uiHandler = new GitBranchUiHandlerImpl(project, indicator);
-      new GitBranchWorker(project, Git.getInstance(), uiHandler)
-          .checkout(/* reference */ gitRebaseParameters.getCurrentBranch().getName(), /* detach */ false,
-              Collections.singletonList(gitRepository));
+    val actionId = new SideEffectiveActionId("rebase");
```

Comment:
probably it will be handy to have the actions ids extracted or use the action classes simple names for that

Reply:
Hmmm I'm using such user-readable strings so that they can be put subsequently put into action description: _Disable due to the ongoing **rebase**_ etc.